nemo/collections/nlp/modules/common/huggingface/huggingface_utils.py

```diff
@@ -27,78 +28,94 @@
     },
-    'roberta': {
-        'default': 'roberta-base',
-        'class': RobertaEncoder,
-        'config': RobertaConfig,
-        'pretrained_model_list': ROBERTA_PRETRAINED_MODEL_ARCHIVE_LIST,
+    ".modeling_roberta.": {
+        "default": "roberta-base",
+        "class": RobertaEncoder,
+        "config": RobertaConfig,
+        "pretrained_model_list": ROBERTA_PRETRAINED_MODEL_ARCHIVE_LIST,
     },
-    'albert': {
-        'default': 'albert-base-v2',
-        'class': AlbertEncoder,
-        'config': AlbertConfig,
-        'pretrained_model_list': ALBERT_PRETRAINED_MODEL_ARCHIVE_LIST,
+    ".modeling_albert.": {
+        "default": "albert-base-v2",
+        "class": AlbertEncoder,
+        "config": AlbertConfig,
+        "pretrained_model_list": ALBERT_PRETRAINED_MODEL_ARCHIVE_LIST,
     },
 }
 
 
 def get_huggingface_lm_model(
     pretrained_model_name: str, config_dict: Optional[dict] = None, config_file: Optional[str] = None,
 ):
-    '''
+    """
     Returns lm model instantiated with Huggingface
+
     Args:
         pretrained_mode_name: specify this to instantiate pretrained model from Huggingface,
             e.g. bert-base-cased. For entire list, see get_huggingface_pretrained_lm_models_list().
         config_dict: model configuration dictionary used to instantiate Huggingface model from scratch
         config_file: path to model configuration file used to instantiate Huggingface model from scratch
-    '''
-
-    # assumes that model configuration is correct, otherwise will pass on Huggingface's model instantiation error
-    model_type = pretrained_model_name.split("-")[0]
-    model_class = HUGGINGFACE_MODELS[model_type]['class']
-    if config_file:
-        config_class = HUGGINGFACE_MODELS[model_type]['config']
-        return model_class(config_class.from_json_file(config_file))
-    elif config_dict:
-        config_class = HUGGINGFACE_MODELS[model_type]['config']
-        return model_class(config=config_class(**config_dict))
+
+    Returns:
+        BertModule
+    """
+
+    # assumes that model configuration is correct, otherwise will pass on Huggingface"s model instantiation error
+    automodel = AutoEncoder(pretrained_model_name)
+
+    model_type = re.search(r'.modeling_[A-z]*[0-5]?\.', str(automodel.type))[0]
+
+    if model_type in HUGGINGFACE_MODELS:
+        model_class = HUGGINGFACE_MODELS[model_type]["class"]
+        if config_file:
+            config_class = HUGGINGFACE_MODELS[model_type]["config"]
+            return model_class(config_class.from_json_file(config_file))
+        elif config_dict:
+            config_class = HUGGINGFACE_MODELS[model_type]["config"]
+            return model_class(config=config_class(**config_dict))
+        else:
+            return model_class.from_pretrained(pretrained_model_name)
     else:
-        return model_class.from_pretrained(pretrained_model_name)
+        logging.info('Using HuggingFace AutoModel, typing check is not supported')
```

Comment:
change to "model type" is not supported or "model class". Typing check is confusing